elf_load.c

```diff
@@ -285,11 +285,11 @@ struct elf_prog *elf_load_img(struct uk_alloc *a, void *img_base,
 		 * Setup memory protection
 		 */
 		if (pt) {
-			uk_pr_debug("%p: Protecting 0x%"PRIx64" - 0x%"PRIx64": %c%c%c\n",
+			uk_pr_debug("%p: Protecting 0x%llx - 0x%llx: %c%c%c\n",
 				    img_base,
-				    ALIGN_DOWN((uint64_t) (elf_prog->img
+				    (unsigned long long) ALIGN_DOWN((uint64_t) (elf_prog->img
```

Comment:
Hum... weird that there is actually a problem occuring. `"%"PRIx64` is the official libc provided format definition to print a `uint64_t`/`int64_t` correctly. What is the resulting type of `ALIGN_UP`/`ALIGN_DOWN`? I guess the `phdr` fields use `libelf` defined 64bit equivalents. It is possible that they use `long long` instead of `long`. On x86_64 for Linux and us, they are same size but it could be that the compiler is still throws a warning. I'd prefer if we cast the second argument of the macro to `uint64_t`, or we cast the result to `uint64_t` instead (instead of `(unsigned long long)`). This would be more consistent with the rest of the code within this file.

Reply:
@FotisKouzinos can you also add a very short description in the commit message that tells what the code is fixing?